tests/garbage_collection/gc_delete_test_coop.cpp

```diff
@@ -0,0 +1,169 @@
+#include "backend/planner/seq_scan_plan.h"
+#include "backend/planner/update_plan.h"
+
+using ::testing::NotNull;
+using ::testing::Return;
+
+namespace peloton {
+namespace test {
+
+//===--------------------------------------------------------------------===//
+// GC Tests
+//===--------------------------------------------------------------------===//
+
+class GCDeleteTestCoop : public PelotonTest {};
+
+std::atomic<int> tuple_id;
+std::atomic<int> delete_tuple_id;
+enum GCType type = GC_TYPE_COOPERATIVE;
+
+void InsertTuple(storage::DataTable *table, VarlenPool *pool) {
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  std::unique_ptr<executor::ExecutorContext> context(
+      new executor::ExecutorContext(txn));
+
+  for (oid_t tuple_itr = 0; tuple_itr < 10; tuple_itr++) {
+    auto tuple = ExecutorTestsUtil::GetTuple(table, ++tuple_id, pool);
+
+    planner::InsertPlan node(table, std::move(tuple));
+    executor::InsertExecutor executor(&node, context.get());
+    executor.Execute();
+  }
+
+  txn_manager.CommitTransaction();
+}
+
+void DeleteTuple(storage::DataTable *table) {
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  std::unique_ptr<executor::ExecutorContext> context(
+      new executor::ExecutorContext(txn));
+
+  std::vector<storage::Tuple *> tuples;
+
+  // Delete
+  planner::DeletePlan delete_node(table, false);
+  executor::DeleteExecutor delete_executor(&delete_node, context.get());
+
+  // Predicate
+
+  // WHERE ATTR_0 > 60
+  expression::TupleValueExpression *tup_val_exp =
+      new expression::TupleValueExpression(0, 0);
+  expression::ConstantValueExpression *const_val_exp =
+      new expression::ConstantValueExpression(
+          ValueFactory::GetIntegerValue(60));
+  auto predicate = new expression::ComparisonExpression<expression::CmpGt>(
+      EXPRESSION_TYPE_COMPARE_GREATERTHAN, tup_val_exp, const_val_exp);
+
+  // Seq scan
+  std::vector<oid_t> column_ids = {0};
+  std::unique_ptr<planner::SeqScanPlan> seq_scan_node(
+      new planner::SeqScanPlan(table, predicate, column_ids));
+  executor::SeqScanExecutor seq_scan_executor(seq_scan_node.get(),
+                                              context.get());
+
+  // Parent-Child relationship
+  delete_node.AddChild(std::move(seq_scan_node));
+  delete_executor.AddChild(&seq_scan_executor);
+
+  EXPECT_TRUE(delete_executor.Init());
+  EXPECT_TRUE(delete_executor.Execute());
+  // EXPECT_TRUE(delete_executor.Execute());
+
+  txn_manager.CommitTransaction();
+}
+
+TEST_F(GCDeleteTestCoop, DeleteTest) {
+
+  peloton::gc::GCManagerFactory::Configure(type);
+  peloton::gc::GCManagerFactory::GetInstance().StartGC();
+
+  auto *table = ExecutorTestsUtil::CreateTable(1024);
+  auto &manager = catalog::Manager::GetInstance();
+  storage::Database db(DEFAULT_DB_ID);
+  manager.AddDatabase(&db);
```

Comment:
If you use this way to create a new db in your test, probably you should also add `manager.DropDatabaseWithOid(current_db_oid); ` 
In the end of this tests.

We meet similar problem that you meet during our development: a single test success, but when run together with others, it fails. Later we find that it is because we didn't clean the data of temporary created database clearly, and after add this line in the end of our test, it works perfectly well. You can refer to line 892 in our tests/catalog/constraints_test.cpp and have a try. Hope this will help you